packages/aggregator/src/analytics/aggregateAnalytics/aggregations/offsetPeriod.js

```diff
@@ -24,31 +31,37 @@ const validateOffsetPeriodConfig = (config = {}) => {
 
 const addOffsetToPeriod = (period, { periodType, offset }) => {
   const momentUnit = periodTypeToMomentUnit(periodType.toUpperCase());
-  const momentWithOffset = utcMoment(period).add(offset, momentUnit);
-  const currentPeriodType = periodToType(period);
-  return momentToPeriod(momentWithOffset, currentPeriodType);
+  const momentWithOffset = periodToMoment(period).add(offset, momentUnit);
+  return momentToPeriod(momentWithOffset, periodToType(period));
 };
 
-export const offsetPeriod = (analytics, aggregationConfig) => {
-  validateOffsetPeriodConfig(aggregationConfig);
+export const offsetPeriod = (analytics, config) => {
+  validateOffsetPeriodConfig(config);
 
   return analytics.map(analytic => ({
     ...analytic,
-    period: addOffsetToPeriod(analytic.period, aggregationConfig),
+    period: addOffsetToPeriod(analytic.period, config),
   }));
 };
 
-/**
- * Expands the provided date to the opposite direction than the offset
- * so that enough data will be available for its calculation
- */
-const compensateForPeriodOffset = (date, { periodType, offset }) =>
-  addOffsetToPeriod(date, { periodType, offset: offset * -1 });
-
 export const getDateRangeForOffsetPeriod = (dateRange, config) => {
   validateOffsetPeriodConfig(config);
+
+  const periodType = config.periodType.toUpperCase();
+  // Expands the provided date to the opposite direction than the offset
+  // so that enough data will be available for its calculation
+  const offsetCompensation = config.offset * -1;
+
+  const getMomentWithOffset = dateString => {
+    const momentUnit = periodTypeToMomentUnit(periodType);
+    return utcMoment(dateString).add(offsetCompensation, momentUnit);
+  };
+
+  const startMoment = getMomentWithOffset(dateRange.startDate);
+  const endMoment = getMomentWithOffset(dateRange.endDate).endOf(periodType);
```

Comment:
Interested in what this does when going from e.g. `2020-01-31` with an offset of 1 so we're wanting to include data up to `2020-02-28`. Worried it would end up at `2020-03-31`. You've got a good edge case test for going from a smaller month to a larger one, but how about a larger to a smaller?

Reply:
The behaviour going from larger to smaller is already tested in [offsetPeriod tests](https://github.com/beyondessential/tupaia/blob/dev/packages/aggregator/src/__tests__/analytics/aggregateAnalytics/aggregations/offsetPeriod.test.js#L167). Based on [this comment](https://github.com/beyondessential/tupaia/blob/dev/packages/aggregator/src/__tests__/analytics/aggregateAnalytics/aggregations/offsetPeriod.test.js#L193), I am only testing this edge case cause going from smaller to larger is different for periods:
`20190430 + 1 month` => `20190530`

and date strings:
`2019-04-30 + 1 month` => `2019-05-30`

I personally it's fine as is, but let me know if you think it's a bit unclear, and I can add a smaller to larger test case for `getDateRangeForOffsetPeriod` too

Reply:
Hey but I just realised the two functions have now become separated, as they should (since the one uses periods, and the other date strings)!! So I'll update the tests here to include a complete period type suite. Nice catch!